topology/probes/k8s/namespace.go

```diff
@@ -62,43 +56,16 @@ func newNamespaceProbe(clientset *kubernetes.Clientset, g *graph.Graph) Subprobe
 	return NewResourceCache(clientset.Core().RESTClient(), &v1.Namespace{}, "namespaces", g, &namespaceHandler{})
 }
 
-func newNamespaceLinker(g *graph.Graph, subprobes map[string]Subprobe) probe.Probe {
-	nsSubprobe := subprobes["namespace"]
-	if nsSubprobe == nil {
+func newNamespaceLinker(g *graph.Graph, subprobes map[string]Subprobe, manager, ty string) probe.Probe {
+	dstCache := subprobes[ty]
+	if dstCache == nil {
 		return nil
 	}
 
-	filter := filters.NewAndFilter(
-		filters.NewTermStringFilter("Manager", Manager),
-		filters.NewTermStringFilter("Type", "namespace"),
-		filters.NewNotNullFilter("Name"),
-	)
-
-	namespaceIndexer := graph.NewMetadataIndexer(g, nsSubprobe, graph.NewElementFilter(filter), "Name")
+	namespaceIndexer := graph.NewMetadataIndexer(g, g, graph.Metadata{"Manager": Manager, "Type": "namespace"}, "Name")
```

Comment:
This creates a namespace indexer per resource type. This is very costly

Reply:
see above (my comment on the cluster indexer)  - this simplifies implementation and no apparent execution cost.

Reply:
Did you try this on a platform with hundred or thousand of namespaces ? I really don't see the difference in terms of implementation between creating one indexer will a big filter and creating many indexers and iterating over a loop. Both a almost equal in terms of complexity but one has clearly a bigger execution cost

Reply:
I don't see the difference in performance between the two approaches (as it is we have never tested our system for thousands of namespaces, if the probe breaks it will not be due to the above change ...).

and again the change towards a separate linker per each namespace--><resource> (e.g. namespace-->pod; namespace-->service) was triggered by the need to add namespace--><istio resource> links thus suggesting that each namespace--><resource> linker must work independantly.

Reply:
Before your change:
- Only one indexer was used to index namespaces
- Only one indexer was used to index the resources that must be linked to namespaces

After your change:
- One indexer is used to index namespace **per resource type** (let's say there is 10 resource types. For every change on the graph, 10 methods will be called, 10 hash will be computed and 10 entries will be created in indexes)
- One indexer is used per resource type (same calculus than above)

I don't understand what you mean by `was triggered by the need to add namespace--> links`. What is namespace -> links ?

Reply:
in the previous implementation all the following were handled by a single linker:
- namespace --> pod
- namespace --> networkpolicy
- namespace --> etc.

but as we now also have the ability of istio probe to link its' resources under namespaces I wanted to expose the ability to link individual resources under namespace such as:
- namespace (k8s) --> destinationrule (istio)

this resulting implementation has now much simplified the namespace linker and I still don't see the significant impact on performance (yes adding/deleting/editing namespace resources will now  echo on 10 indexers (one per resource) rather than a single indexer ... but namespaces are small in number compared to the entire k8s landscape (1/100) and events on namespaces are very rare (compared to pods for example ...) so this is not a matter of true performance impact thus I prefer to go with the simpler implementation.

Reply:
If the goal is to be able to create `namespace (k8s) --> destinationrule (istio)` (and I understand this requirement), we can think of an other way.

Now, I really don't like to say it like this but I am not going to merge this PR as we don't seem to be able to agree on it.